src/comm/MAVLinkProtocol.cc

```diff
@@ -197,7 +197,8 @@ void MAVLinkProtocol::receiveBytes(LinkInterface* link, QByteArray b)
     // Since receiveBytes signals cross threads we can end up with signals in the queue
     // that come through after the link is disconnected. For these we just drop the data
     // since the link is closed.
-    if (!_linkMgr->containsLink(link)) {
```

Comment:
If you keep both the containsLinks and the weak pointer check then I believe the null ptr return from LinkManager can stay a qWarning.

Reply:
I'm not a fan 2 calls or iterating the list twice (even though it's a small one), so I added a flag.